openstates/nc/bills.py

```diff
@@ -258,14 +288,172 @@ def scrape_votes(self, bill, doc):
+                            )
+                            r_number = vote_details_line[21:23]
+                            action_number = vote_details_line[25:28]
+                            cod = vote_details_line[29:33]
+
+                            # Determining how the bill was voted
+                            vote_numbers = vote_details_line.split()
+                            yes_votes = int(vote_numbers[-5])
+                            no_votes = int(vote_numbers[-4])
+                            noVt_votes = int(vote_numbers[-3])
+                            exAb_votes = int(vote_numbers[-2])
+                            exVt_votes = int(vote_numbers[-1])
+                            if yes_votes > (
+                                no_votes + noVt_votes + exAb_votes + exVt_votes
+                            ):
+                                action_vote_result = "pass"
+                            else:
+                                action_vote_result = "fail"
+
+                            rep_vote = vote_text[x + y + 1]
+                            vote_location = rep_vote.rfind("X")
+
+                            if vote_location == yes_location:
+                                how_voted = "yes"
+                            elif vote_location == no_location:
+                                how_voted = "no"
+                            elif vote_location == noVt_location:
+                                how_voted = "other"
+                            elif vote_location == exAb_location:
+                                how_voted == "absent"
+                            else:
+                                how_voted = "excused"
+
+                            # print("Bill ID", bill_id, "How Voted:", how_voted)
+                            archived_votes[bill_id][
+                                (
+                                    vote_date,
+                                    r_number,
+                                    action_number,
+                                    action_vote_result,
+                                    archive_url,
+                                    cod,
+                                    chamber,
+                                )
+                            ].append({"leg": rep_name, "how_voted": how_voted})
+                        else:
+                            break
+
     def scrape(self, session=None, chamber=None):
         if not session:
             session = self.latest_session()
             self.info("no session specified, using %s", session)
 
         chambers = [chamber] if chamber else ["upper", "lower"]
+
+        if session in ["1997", "1999"]:
+            self.scrape_archived_votes("upper", session)
+            self.scrape_archived_votes("lower", session)
+
         for chamber in chambers:
-            yield from self.scrape_chamber(chamber, session)
+
+            if session in ["1997", "1999"]:
```

Comment:
this if statement doesn't do anything anymore :)

Reply:
Removed!